simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -468,28 +776,318 @@ def __init__(
+    according to:
+
+    .. math::
+        \begin{bmatrix} T_{xx} & T_{yx} & T_{zx} \\ T_{xy} & T_{yy} & T_{zy} \end{bmatrix} =
+        \begin{bmatrix} H_x^{(x)} & H_y^{(x)} \\ H_x^{(y)} & H_y^{(y)} \end{bmatrix}_b^{-1} \,
+        \begin{bmatrix} H_x^{(x)} & H_y^{(x)} & H_z^{(x)} \\ H_x^{(y)} & H_y^{(y)} & H_z^{(y)} \end{bmatrix}_r
+
+    where subscript :math:`b` denotes the base station location and subscript
+    :math:`r` denotes the mobile receiver location.
+
+    Parameters
+    ----------
+    locations : (n_loc, n_dim) numpy.ndarray
+        Mobile receiver locations.
+    orientation : {'xx', 'yx', 'zx', 'zy', 'yy', 'zy'}
+        Specifies the tipper element :math:`T_{ij}` corresponding to the data.
+    component : {'real', 'imag'}
+        Tipper data type. For the tipper element :math:`T_{ij}` specified by the `orientation`
+        input argument, the receiver can be set to compute the following:
+        - 'real': Real component of the tipper (unitless)
+        - 'imag': Imaginary component of the tipper (unitless)
+    locations_bs : (n_loc, n_dim) numpy.ndarray, optional
+        Locations for remote horizontal magnetic field measurements (e.g. base station).
+        Must be same shape as `locations`.
+    locations_e : (n_loc, n_dim) numpy.ndarray, optional
+        Deprecated property. To be removed in simpeg 0.20.0.
+    locations_h : (n_loc, n_dim) numpy.ndarray, optional
+        Deprecated property. To be removed in simpeg 0.20.0.
+    """
+
+    def __init__(
+        self,
+        locations,
+        orientation="zx",
+        component="real",
+        locations_bs=None,
+        locations_e=None,
+        locations_h=None,
+    ):
+
+        self.orientation = orientation
+        self.component = component
+
+        # check if locations and locations_bs have been provided
+        if locations_bs is not None:
+            # check that locations are same size
+            if locations_bs.size == locations.size:
+                self._locations_bs = locations_bs
+            else:
+                raise Exception("location_bs needs to be same size as locations")
+
+        else:
+            # check shape of locations
+            if isinstance(locations, list):
+                if len(locations) == 1:
+                    pass
+                elif len(locations) == 2:
+                    if locations[0].size != locations[1].size:
+                        raise Exception(
+                            "location_bs needs to be same size as locations"
+                        )
+                else:
+                    raise Exception("incorrect size of list, must be length of 1 or 2")
+
+                self._locations_bs = locations[0]
+                locations = locations[-1]
+            elif isinstance(locations, np.ndarray):
+                self._locations_bs = locations
+            else:
+                raise Exception("locations need to be either a list or numpy array")
```

Comment:
It's better to specify the type of exception instead of using `Exception`.
```suggestion
                raise TypeError("locations need to be either a list or numpy array")
```